File: kimhyunji/PGS/Level 1/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = {7, 7};

        for(int i=0; i<lottos.length; i++){
            if(lottos[i] == 0) answer[0]--;
            for(int j=0; j<win_nums.length; j++){
                if(lottos[i] == win_nums[j]){
                    answer[0]--;
                    answer[1]--;
                }
            }
        }

        if(answer[0] < 1) answer[0] = 1;
        if(answer[0] > 6) answer[0] = 6;
        if(answer[1] > 6) answer[1] = 6;

        return answer;
    }
}

File: baekyeonsu/PGS/Lv. 1/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
  const validations = lottos.filter((number) => {
    if (number !== 0) return number;
  });
  const unknownNumber = lottos.length - validations.length;
  let result = new Array(2).fill(6);

  if (unknownNumber === 6) {
    result[0] -= unknownNumber - 1;
    return result;
  }

  let same = 0;
  for (let i = 0; i < validations.length; i++) {
    if (win_nums.includes(validations[i])) {
      same += 1;
      if (same >= 2) {
        result[0] -= 1;
        result[1] -= 1;
      }
    }
  }

  if (unknownNumber + same >= 2) result[0] -= unknownNumber;
  return result;
}
